nixos/modules/services/web-servers/unit/default.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.unit;
+
+  configFile = builtins.toFile "unit.json" ''
```

Comment:
In a recent PR you did, [I mentioned](https://github.com/NixOS/nixpkgs/pull/56280#discussion_r259626780) how you could use `builtins.toFile` if the file is constant. That wasn't entirely right, this should only be used if the file doesn't depend on any derivations (and can in fact only be used then).

Anyways, this is not the case here, the `config` option supplied by the user may very well depend on a derivation, so you should use `pkgs.writeText` instead here.